### ISSUE TITLE
WDM appears to incorrectly pick v78.x chrome driver when v77 Chrome exists on the system

### ISSUE THREAD

**Issue by deepakab03** [bonigarcia/webdrivermanager#394]
**Description of the problem**: WDM was being used on a number of build nodes that had Chrome v77.0.3865.120 (Official Build) (64-bit) and it was downloading the corresponding web-driver (version 77.0.3865.120)  and the same was being used to run tests. Yesterday, builds started failing as WDM started to download v 78.0.3904.70 but the Chrome was still at v77.x. When Chrome in each system was manually updated (Open Chrome, do Help -> About) the build(s) passed.
Question - What triggered WDM to use v 78.0.3904.70 when Chrome browser was at v77.0.3865.120? How do we prevent this from happening in future?

**Browser and version**: Chrome 77.0.3865.120. 

**Operating system**: Windows server 2012 R2 64 bit

**WebDriverManager version**: 3.7.0 - **also tested with 3.7.1 - same issue**

**WebDriverManager use**: launching standard Selenium tests

**WebDriverManager traces**: 
02:12:57  [main] I| 16:42:57.939 | [i.g.b.wdm.WebDriverManager:1037] | Reading https://chromedriver.storage.googleapis.com/ to seek chromedriver
02:12:57  [main] I| 16:42:58.350 | [i.g.bonigarcia.wdm.Downloader:128] | Downloading https://chromedriver.storage.googleapis.com/78.0.3904.70/chromedriver_win32.zip
02:12:58  [main] I| 16:42:59.442 | [i.g.bonigarcia.wdm.Downloader:183] | Extracting binary from compressed file chromedriver_win32.zip
02:12:59  [main] I| 16:42:59.901 | [i.g.b.wdm.WebDriverManager:1073] | Exporting webdriver.chrome.driver as C:\XXX\.m2\repository\webdriver\chromedriver\win32\78.0.3904.70\chromedriver.exe

**Error trace**: 
02:13:01  session not created: This version of ChromeDriver only supports Chrome version 78
02:13:01  Build info: version: '3.14.0', revision: 'aacccce0', time: '2018-08-02T20:19:58.91Z'
02:13:01  System info: host: 'XXXX', ip: 'XXXX', os.name: 'Windows Server 2012 R2', os.arch: 'amd64', os.version: '6.3', java.version: '1.8.0_162'
02:13:01  Driver info: driver.version: ChromeDriver
02:13:01  remote stacktrace: Backtrace:
02:13:01  	Ordinal0 [0x00CBA903+1550595]
02:13:01  	Ordinal0 [0x00C3A701+1025793]
02:13:01  	Ordinal0 [0x00BBC6E5+509669]
02:13:01  	Ordinal0 [0x00B4FC68+64616]
02:13:01  	Ordinal0 [0x00B4C594+50580]
02:13:01  	Ordinal0 [0x00B6ACE7+175335]
02:13:01  	Ordinal0 [0x00B6A8ED+174317]
02:13:01  	Ordinal0 [0x00B68CDB+167131]
02:13:01  	Ordinal0 [0x00B5144A+70730]
02:13:01  	Ordinal0 [0x00B524D0+74960]
02:13:01  	Ordinal0 [0x00B52469+74857]
02:13:01  	Ordinal0 [0x00C542C7+1131207]
02:13:01  	GetHandleVerifier [0x00D570FD+523789]
02:13:01  	GetHandleVerifier [0x00D56E90+523168]
02:13:01  	GetHandleVerifier [0x00D5E1E7+552695]
02:13:01  	GetHandleVerifier [0x00D578FA+525834]
02:13:01  	Ordinal0 [0x00C4B7FC+1095676]
02:13:01  	Ordinal0 [0x00C5633B+1139515]
02:13:01  	Ordinal0 [0x00C564A3+1139875]
02:13:01  	Ordinal0 [0x00C55425+1135653]
02:13:01  	BaseThreadInitThunk [0x763B7C04+36]
02:13:01  	RtlInitializeExceptionChain [0x77DDAB8F+143]
02:13:01  	RtlInitializeExceptionChain [0x77DDAB5A+90]

**Comment by bonigarcia** [comment]
Is this issue still happening to you? If so, please post your WebDriverManager traces also at DEBUG level.

**Comment by deepakab03** [comment]
Hello,
Sorry for not responding earlier, lost the email update. 
I couldn't get any logs to you since the issue went away with the Chrome browser update (done manually)
We faced the same issue again yesterday: 

1. Google released Stable chrome for desktop yesterday - 4th Feb
2. Using the same WDM v3.7.0., without any changes the build - on build nodes only -  started to fail with the message: _This version of ChromeDriver only supports Chrome version 80_ 
3. Checking the Chrome version on the build node both via the registry HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon as well as in the browser: Chrome -> Help -> About, gave the previous Chrome version: v79.03945.130. Of course as soon as you check the Chrome version via the browser it updates itself to v80, so the issue can't be reproduced there!!

Logs:

```
Running on symmwinbuildw95.cec.lab.emc.com 
04:09:28  [main] I| 17:39:30.698 | [c.e.s.t.s.mgr.WebDriverFacade:38] | Using webDriver 'chrome' with headless mode 'true'
04:09:30  [main] I| 17:39:32.700 | [i.g.b.wdm.WebDriverManager:1037] | Reading https://chromedriver.storage.googleapis.com/ to seek chromedriver
04:09:31  [main] I| 17:39:33.777 | [i.g.bonigarcia.wdm.Downloader:128] | Downloading https://chromedriver.storage.googleapis.com/80.0.3987.16/chromedriver_win32.zip
04:09:37  [main] I| 17:39:39.126 | [i.g.bonigarcia.wdm.Downloader:183] | Extracting binary from compressed file chromedriver_win32.zip
04:09:37  [main] I| 17:39:39.593 | [i.g.b.wdm.WebDriverManager:1073] | Exporting webdriver.chrome.driver as C:\Users\<blah-user>\.m2\repository\webdriver\chromedriver\win32\80.0.3987.16\chromedriver.exe
04:09:39  Starting ChromeDriver 80.0.3987.16 (320f6526c1632ad4f205ebce69b99a062ed78647-refs/branch-heads/3987@{#185}) on port 21064
04:09:39  Only local connections are allowed.
04:09:39  Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
04:09:41  [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 13.419 s <<< FAILURE! - in com.abc.DbeIT
04:09:41  [ERROR] Feature: Check important features of Inlines module work  Time elapsed: 13.418 s  <<< ERROR!
04:09:41  org.openqa.selenium.SessionNotCreatedException: 
04:09:41  session not created: This version of ChromeDriver only supports Chrome version 80
04:09:41  Build info: version: '3.14.0', revision: 'aacccce0', time: '2018-08-02T20:19:58.91Z'
04:09:41  System info: host: 'ABCHOST', ip: 'x.y.z.d', os.name: 'Windows Server 2012 R2', os.arch: 'amd64', os.version: '6.3', java.version: '1.8.0_162'
04:09:41  Driver info: driver.version: ChromeDriver
04:09:41  remote stacktrace: Backtrace:
```

It appears the WDM on the Windows build node is not able to find the Chrome version, hence defaults to downloading latest version. I tried running _java -jar webdrivermanager-3.8.1-fat.jar chrome_ and got the following output:

```
[INFO] Using WebDriverManager to resolve chrome
[DEBUG] Running command on the shell: [wmic, datafile, where, name='C:\\Program Files (x86)\\Google\
\Chrome\\Application\\chrome.exe', get, Version, /value]
[DEBUG] There was a problem executing command <wmic datafile where name='C:\\Program Files (x86)\\Go
ogle\\Chrome\\Application\\chrome.exe' get Version /value> on the shell: Cannot run program "wmic" (
in directory "C:\Windows\System32"): CreateProcess error=2, The system cannot find the file specifie
d
[DEBUG] Result:
[DEBUG] Running command on the shell: [wmic, datafile, where, name='C:\\Users\\AbcUser\\AppDat
a\\Local\\Google\\Chrome\\Application\\chrome.exe', get, Version, /value]
[DEBUG] There was a problem executing command <wmic datafile where name='C:\\Users\\AbcUser\\A
ppData\\Local\\Google\\Chrome\\Application\\chrome.exe' get Version /value> on the shell: Cannot run
 program "wmic" (in directory "C:\Windows\System32"): CreateProcess error=2, The system cannot find
the file specified
[DEBUG] Result:
[DEBUG] Running command on the shell: [wmic, datafile, where, name='C:\\Program Files\\Google\\Chrom
e\\Application\\chrome.exe', get, Version, /value]
[DEBUG] There was a problem executing command <wmic datafile where name='C:\\Program Files\\Google\\
Chrome\\Application\\chrome.exe' get Version /value> on the shell: Cannot run program "wmic" (in dir
ectory "C:\Windows\System32"): CreateProcess error=2, The system cannot find the file specified
[DEBUG] Result:
[DEBUG] The proper chromedriver version for your Google Chrome is unknown ... trying with the latest

[DEBUG] Latest version of chromedriver according to https://chromedriver.storage.googleapis.com/LATE
ST_RELEASE is 80.0.3987.16
[INFO] Reading https://chromedriver.storage.googleapis.com/ to seek chromedriver
[INFO] Downloading https://chromedriver.storage.googleapis.com/80.0.3987.16/chromedriver_win32.zip
[INFO] Extracting binary from compressed file chromedriver_win32.zip
[INFO] Resulting binary C:\tmp\chromedriver.exe

```

Is it possible to enhance web-driver manager on windows so that in case wmic is not available, the _HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon_ Registry key can be tried to find the Chrome version that is present?

**Comment by bonigarcia** [comment]
I think it could be done. I'll try to implement it for a future version.

**Comment by deepakab03** [comment]
Great! That's fantastic, thank you for spending time to create this very useful tool and taking such requests!

**Comment by bonigarcia** [comment]
This feature is available in WebDriverManager 4.0.0.
